### ISSUE TITLE
Compilation failure on Titan when using gasnet

### ISSUE THREAD

**Issue by sdalton1** [StanfordLegion/legion#181]
Compilation fails on Titan when USE_GASNET is enabled. 

Stripped down command that fails
```shell
CC -o /ccs/home/sdalton/legion/runtime/legion/legion_views.o -c /ccs/home/sdalton/legion/runtime/legion/legion_views.cc  -DGASNETI_BUG1389_WORKAROUND=1 -DUSE_GASNET -I/ccs/home/sdalton/legion/runtime -I/ccs/home/sdalton/legion/runtime/realm -I/ccs/home/sdalton/legion/runtime/legion -I/ccs/home/sdalton/legion/runtime/mappers -I/ccs/home/sdalton/gasnet/release/include -I/ccs/home/sdalton/gasnet/release/include/gemini-conduit
```

Debug session with Mike revealed that one of the gasnet headers is defining a conflicting macro before the legion_utilities.h file is included.

```c++
#define __STDC_FORMAT_MACROS
#include <inttypes.h>
#define MASK_FMT "%16.16" PRIx64
```

```shell
/ccs/home/sdalton/legion/runtime/legion/legion_utilities.h: In member function 'char* Legion::BitMask<T, MAX, SHIFT, MASK>::to_string() const':
/ccs/home/sdalton/legion/runtime/legion/legion_utilities.h:49:27: error: expected ')' before 'PRIx64'
 #define MASK_FMT "%16.16" PRIx64
```

**Comment by lightsighter** [comment]
The problem is GASNet is including inttypes.h before this one in the include hierarchy for legion_utilities.h, but they are not prefixing it with the __STDC_FORMAT_MACROS define.  As a result the PRIx64 macro is not being defined.

**Comment by streichler** [comment]
@sdalton1 can you try adding -D__STDC_FORMAT_MACROS to your CC_FLAGS?  If that works, we'll jam it into the standard CC_FLAGS - we've seen one other system that needs this too.

**Comment by sdalton1** [comment]
Adding __STDC_FORMAT_MACROS fixes the issue. The only side effect is a warning about redefining __STDC_FORMAT_MACROS.

```shell
/ccs/home/sdalton/legion/runtime/legion/legion_utilities.h:48:0: warning: "__STDC_FORMAT_MACROS" redefined
 #define __STDC_FORMAT_MACROS
```

**Comment by streichler** [comment]
@sdalton1 This should hopefully be fixed (sans warning) with 208a21d.

**Comment by sdalton1** [comment]
Issue resolved. Closing.
